### PR TITLE
Implement NATS PubSub Destination

### PR DESCRIPTION
**Pull request by alyakimenko**
### Description

Implement NATS PubSub Destination.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-s3/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,3 +51,26 @@ The config passed to Configure can contain the following fields.
 | `maxReconnects`            | Sets the number of NATS server reconnect attempts that will be tried before giving up. If negative, then it will never give up trying to reconnect.                                                                                               | false    | `5`                                |
 | `reconnectWait`            | Sets the time to backoff after attempting a reconnect to a NATS server that the connector was already connected to previously.                                                                                                                    | false    | `5s`                               |
 | `bufferSize`               | A buffer size for consumed messages. It must be set to avoid the [slow consumers](https://docs.nats.io/running-a-nats-service/nats_admin/slow_consumers) problem. Minimum allowed value is `64`                                                   | false    | `1024`                             |
+
+## Destination
+
+### Sending messages
+
+The connector sends message synchronously, one by one.
```
Comment (hariso) [inline README.md:59]: Is it possible to send message async?

File: acceptance_test.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			Config: sdk.ConfigurableAcceptanceTestDriverConfig{
+				Connector:         Connector,
+				SourceConfig:      cfg,
+				DestinationConfig: cfg,
+				BeforeTest: func(t *testing.T) {
+					subject := t.Name() + uuid.New().String()
+					cfg[config.KeySubject] = subject
+				},
+				GoleakOptions: []goleak.Option{
+					// nats.go spawns a separate goroutine to process flush requests
```
Comment (hariso) [inline acceptance_test.go:86]: Is it possible to stop those routines?

File: README.md [hunk 3]
```diff
@@ -51,3 +51,26 @@ The config passed to Configure can contain the following fields.
...
+| `tls.clientCertPath`       | A path pointed to a TLS client certificate, must be present if tls.clientPrivateKeyPath field is also present. Must be a valid file path. Required if your NATS server is using TLS.                                                              | false    |                                    |
+| `tls.clientPrivateKeyPath` | A path pointed to a TLS client private key, must be present if tls.clientCertPath field is also present. Must be a valid file path. Required if your NATS server is using TLS.                                                                    | false    |                                    |
+| `tls.rootCACertPath`       | A path pointed to a TLS root certificate, provide if you want to verify server’s identity. Must be a valid file path                                                                                                                              | false    |                                    |
```
Comment (hariso) [inline README.md:74]: Can all of these be used at the same time?

File: destination/destination_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	"github.com/conduitio-labs/conduit-connector-nats-pubsub/config"
+	"github.com/conduitio-labs/conduit-connector-nats-pubsub/test"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+)
+
+func TestDestination_Open(t *testing.T) {
+	t.Parallel()
+
+	t.Run("success, pubsub", func(t *testing.T) {
```
Comment (hariso) [inline destination/destination_integration_test.go:30]: Maybe run all of those below as individual tests test functions?

File: README.md (hunk 1 above)
Comment (alyakimenko) [reply README.md:59]: No, it's not with PubSub. The nats.go library locks any writes to a NATS server and unlocks only after getting a result from a server.

File: destination/destination_integration_test.go (hunk 4 above)
Comment (alyakimenko) [reply destination/destination_integration_test.go]: Yeah, okay. I split those subtests into separate test functions.

File: acceptance_test.go (hunk 2 above)
Comment (hariso) [reply acceptance_test.go:86]: Bump ^

File: destination/destination_integration_test.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+	err = destination.Open(context.Background())
+	is.NoErr(err)
+
+	for i := 0; i < 1000; i++ {
+		err = destination.Write(context.Background(), sdk.Record{
+			Payload: sdk.RawData([]byte("hello")),
+		})
+		is.NoErr(err)
+	}
```
Comment (hariso) [inline destination/destination_integration_test.go:113]: How do we check that this data has actually been written to NATS?

File: acceptance_test.go (hunk 2 above)
Comment (alyakimenko) [reply acceptance_test.go]: I've just checked the acceptance tests, and it seems that these leaks are gone 🤔

Comment (alyakimenko) [reply acceptance_test.go]: I've removed the goleak's options.

File: destination/destination_integration_test.go (hunk 5 above)
Comment (alyakimenko) [reply destination/destination_integration_test.go]: I've added the check

**Review by hariso — APPROVED** [review]

**Pull request merged**